src/main/java/nl/tudelft/lifetiles/graph/controller/GraphController.java

```diff
@@ -54,7 +59,16 @@ public final void initialize(final URL location,
         DefaultGraphParser parser = new DefaultGraphParser();
         Graph<SequenceSegment> gr = parser.parseFile(
                 "data/test_set/simple_graph", gf);
-        return gr;
+        Sequence reference = gr.getSources().iterator().next().getSources()
+                .iterator().next();
```

Comment:
Creating all those new traversers is blasphemy according to the dependency-injection design pattern. You should either use a factory or pass the traversers as abstract type to the method.